libs/parallelism/algorithms/include/hpx/parallel/datapar/iterator_helpers.hpp

```diff
@@ -354,6 +354,44 @@ namespace hpx { namespace parallel { namespace util { namespace detail {
         }
     };
 
+    template <typename V>
+    struct invoke_vectorized_inout1_ind
+    {
+        template <typename F, typename InIter, typename OutIter>
+        HPX_HOST_DEVICE HPX_FORCEINLINE static void call_aligned(
+            F&& f, InIter& it, OutIter& dest)
+        {
+            typedef
+                typename std::iterator_traits<InIter>::value_type value_type;
+
+            V tmp(traits::vector_pack_load<V, value_type>::aligned(it));
+
+            auto ret = HPX_INVOKE(f, tmp);
+            traits::vector_pack_store<decltype(ret), value_type>::aligned(
```

Comment:
Would there be a way to skip the store operation if not necessary (i.e. whenever we know that `f` takes its argument by value)?

Reply:
This is probably not possible generically. Let's leave things as you suggested.